Examples/Io/Root/src/RootTrajectoryParametersWriter.cpp

```diff
@@ -169,71 +169,107 @@ ActsExamples::ProcessCode ActsExamples::RootTrajectoryParametersWriter::writeT(
                                     particleHitCounts);
       if (not particleHitCounts.empty()) {
         // Get the barcode of the majority truth particle
-        m_t_barcode = particleHitCounts.front().particleId.value();
+        long unsigned int barcode =
+            particleHitCounts.front().particleId.value();
+        m_t_barcode.push_back(barcode);
         // Find the truth particle via the barcode
-        auto ip = particles.find(m_t_barcode);
+        auto ip = particles.find(barcode);
         if (ip != particles.end()) {
           const auto& particle = *ip;
-          ACTS_DEBUG("Find the truth particle with barcode = " << m_t_barcode);
+          ACTS_DEBUG("Find the truth particle with barcode = " << barcode);
           // Get the truth particle info at vertex
           const auto p = particle.absoluteMomentum();
-          m_t_charge = particle.charge();
-          m_t_time = particle.time();
-          m_t_vx = particle.position().x();
-          m_t_vy = particle.position().y();
-          m_t_vz = particle.position().z();
-          m_t_px = p * particle.unitDirection().x();
-          m_t_py = p * particle.unitDirection().y();
-          m_t_pz = p * particle.unitDirection().z();
-          m_t_theta = theta(particle.unitDirection());
-          m_t_phi = phi(particle.unitDirection());
-          m_t_eta = eta(particle.unitDirection());
-          m_t_pT = p * perp(particle.unitDirection());
+          m_t_charge.push_back(particle.charge());
+          m_t_time.push_back(particle.time());
+          m_t_vx.push_back(particle.position().x());
+          m_t_vy.push_back(particle.position().y());
+          m_t_vz.push_back(particle.position().z());
+          m_t_px.push_back(p * particle.unitDirection().x());
+          m_t_py.push_back(p * particle.unitDirection().y());
+          m_t_pz.push_back(p * particle.unitDirection().z());
+          m_t_theta.push_back(theta(particle.unitDirection()));
+          m_t_phi.push_back(phi(particle.unitDirection()));
+          m_t_eta.push_back(eta(particle.unitDirection()));
+          m_t_pT.push_back(p * perp(particle.unitDirection()));
         } else {
-          ACTS_WARNING("Truth particle with barcode = " << m_t_barcode
+          ACTS_WARNING("Truth particle with barcode = " << barcode
                                                         << " not found!");
         }
       }
 
       // Get the fitted track parameter
-      m_hasFittedParams = false;
+      bool hasFittedParams = false;
       if (traj.hasTrackParameters(trackTip)) {
-        m_hasFittedParams = true;
+        hasFittedParams = true;
```

Comment:
As you see here, we don't expect that every fitted trajectory is associated to the truth particle. So using the index of the particles to acess the track (`inputFittedTracks[idx]`) as done in the `RootVertexPerformanceWriter.cpp` might not be very appropriate.